internal/controller/purge_controller.go

```diff
@@ -61,60 +61,61 @@ func (r *PurgeReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl
 	kyma := &v1beta2.Kyma{}
 	if err := r.Get(ctx, req.NamespacedName, kyma); err != nil {
 		if util.IsNotFound(err) {
-			logger.V(log.DebugLevel).Info(fmt.Sprintf("Kyma %s not found, probably already deleted",
-				req.NamespacedName))
-			return ctrl.Result{Requeue: false}, nil
+			logger.V(log.DebugLevel).Info(fmt.Sprintf("Kyma %s not found, probably already deleted", req.NamespacedName))
+			return ctrl.Result{}, nil
 		}
-		return ctrl.Result{}, fmt.Errorf("purgeController: %w", err)
+		return ctrl.Result{}, fmt.Errorf("failed getting Kyma %s: %w", req.NamespacedName, err)
 	}
 
 	if kyma.DeletionTimestamp.IsZero() {
-		err := r.ensurePurgeFinalizer(ctx, kyma)
-		if err != nil {
+		if err := r.ensurePurgeFinalizer(ctx, kyma); err != nil {
+			logger.V(log.DebugLevel).Info(fmt.Sprintf("Failed setting purge finalizer for Kyma %s: %s", req.NamespacedName, err))
+			r.raiseSettingPurgeFinalizerFailedEvent(kyma, err)
 			return ctrl.Result{}, err
 		}
-		return ctrl.Result{Requeue: false}, nil
+		return ctrl.Result{}, nil
 	}
 
-	requeueAfter := r.calculateRequeueAfterTime(ctx, kyma)
-	if requeueAfter != 0 {
+	if requeueAfter := r.calculateRequeueAfterTime(kyma); requeueAfter != 0 {
+		logger.Info(fmt.Sprintf("Purge reconciliation for Kyma %s will be requeued after %s", req.NamespacedName, requeueAfter))
```

Comment:
Reviewer note: This message is promoted from Debug to Info. I am not entirely sure if this is necessary, but this should not re-occur frequently anyway and therefore not spam the infrastructure. And having it we have a hint in the logs that purging was started on info level.

Reply:
I think this log not helpful too much, basically it means when the kyma is under delete, but the time is still not reached `PurgeFinalizerTimeout`, it gets re-queued. With this log message, what we can get is only this kyma CR is still pending to do purge, however we still don't know if this kyma has been deleted during this timeout or it's indeed entered into purge process.

What I expected to see is some logs from the actual purge process, for example, if the log recorded the time that it's purge controller removed the purge finalizer before kyma controller did all finalizers cleanup, then we know this purge process works.

Reply:
added with https://github.com/kyma-project/lifecycle-manager/pull/1392/commits/8e705b62df18921f0646675bd6b3ad12b57b6166